### PR TITLE
Webserver entity sorting

### PR DESCRIPTION
**Pull request by RFDarter**
## Description:
Docs for the webserver entity sorting

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#6445

## Checklist:

  - [x] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [ ] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

**Comment by jesserockz** [comment]
@RFDarter Please update this PR to add the config for all entity types that are supported

**Comment by RFDarter** [comment]
@jesserockz so doing it like on the number component is fine then?

**Comment by jesserockz** [comment]
> @jesserockz so doing it like on the number component is fine then?

Yes I think that is fine.

Although I just realised we need to validate that sorting_weight is only specified when using webserver v3

File: components/web_server.rst [hunk 1]
```diff
@@ -10,20 +10,28 @@ The ``web_server`` component creates a simple web server on the node that can be
...
 
-    Web server version 2
-
+**Web server version 3**
+------------------------
+.. figure:: /components/images/web_server-v3.png
+    :align: center
+    :width: 86.0%
 
+
```
Comment (jesserockz) [inline components/web_server.rst]: I think I prefer these as the image labels rather than new headings on the page.
Also the v3 image seems to be broken?

File: components/web_server.rst [hunk 2]
```diff
@@ -64,7 +72,7 @@ Configuration variables:
 - **ota** (*Optional*, boolean): Turn on or off the OTA feature inside webserver. Strongly not suggested without enabled authentication settings. Defaults to ``true``. Cannot be used with the ``esp-idf`` framework.
 - **id** (*Optional*, :ref:`config-id`): Manually specify the ID used for code generation.
 - **local** (*Optional*, boolean): Include supporting javascript locally allowing it to work without internet access. Defaults to ``false``.
-- **version** (*Optional*, string): ``1`` or ``2``. Version 1 displays as a table. Version 2 uses web components and has more functionality. Defaults to ``2``.
+- **version** (*Optional*, string): ``1``, ``2`` or ``3``. Version 1 displays as a table. Version 2 uses web components and has more functionality. Version 3 uses HA-Styling and has even more functionality. Defaults to ``2``.
```
Comment (jesserockz) [inline components/web_server.rst]: This should be its own PR into `current` as it applies to the current released code.

**Review by jesserockz — CHANGES_REQUESTED** [review]

File: components/web_server.rst (hunk 1 above)
Comment (RFDarter) [reply components/web_server.rst]: ok reverted that, but it looks like the caption of a figure is printed after the image which could  cause some confiusion.

the image for v3 seems to be too lange, since only 300px x 300px is allowed, but the ones for v1 and 2 are also larger

File: components/web_server.rst [hunk 3]
```diff
@@ -16,14 +16,21 @@ will take up *a lot* of memory and may decrease stability, especially on ESP8266
 
     Web server version 1
 
-
+
```
Comment (jesserockz) [inline components/web_server.rst]:
```suggestion

```

File: components/web_server.rst [hunk 4]
```diff
@@ -16,14 +16,21 @@ will take up *a lot* of memory and may decrease stability, especially on ESP8266
...
     Web server version 1
 
-
+    
 .. figure:: /components/images/web_server-v2.png
     :align: center
     :width: 86.0%
 
-    Web server version 2
+    Web server version 2
```
Comment (jesserockz) [inline components/web_server.rst]:
```suggestion
    Web server version 2
```

File: components/web_server.rst [hunk 5]
```diff
@@ -16,14 +16,21 @@ will take up *a lot* of memory and may decrease stability, especially on ESP8266
...
 
-    Web server version 2
+    Web server version 2    
+
+
+.. figure:: /components/images/web_server-v3.png
+    :align: center
+    :width: 86.0%
 
+    Web server version 3
```
Comment (jesserockz) [inline components/web_server.rst]:
```suggestion
    Web server version 3
```

File: components/text/index.rst [hunk 6]
```diff
@@ -50,6 +50,7 @@ Configuration variables:
   for a list of available options. Set to ``""`` to remove the default entity category.
 - **mode** (**Required**, string): Defines how the text should be displayed in the frontend.
   One of ``text`` or ``password``.
+- If Webserver enabled, ``web_server_sorting_weight`` can be set. See :ref:`Webserver Entity Sorting <config-webserver-sorting>`.
```
Comment (jesserockz) [inline components/text/index.rst]:
```suggestion
- If Webserver enabled, ``web_server_sorting_weight`` can be set. See :ref:`Webserver Entity Sorting <config-webserver-sorting>`.
```

File: components/select/index.rst [hunk 7]
```diff
@@ -48,6 +48,7 @@ Configuration variables:
   See https://developers.home-assistant.io/docs/core/entity/#generic-properties
   for a list of available options.
   Set to ``""`` to remove the default entity category.
+- If Webserver enabled, ``web_server_sorting_weight`` can be set. See :ref:`Webserver Entity Sorting <config-webserver-sorting>`.
```
Comment (jesserockz) [inline components/select/index.rst]:
```suggestion
- If Webserver enabled, ``web_server_sorting_weight`` can be set. See :ref:`Webserver Entity Sorting <config-webserver-sorting>`.
```

File: components/number/index.rst [hunk 8]
```diff
@@ -57,6 +57,7 @@ Configuration variables:
 - **device_class** (*Optional*, string): The device class for the number.
   See https://www.home-assistant.io/integrations/number/#device-class
   for a list of available options.
+- If Webserver enabled, ``web_server_sorting_weight`` can be set. See :ref:`Webserver Entity Sorting <config-webserver-sorting>`.
```
Comment (jesserockz) [inline components/number/index.rst]:
```suggestion
- If Webserver enabled, ``web_server_sorting_weight`` can be set. See :ref:`Webserver Entity Sorting <config-webserver-sorting>`.
```

File: components/datetime/index.rst [hunk 9]
```diff
@@ -49,6 +49,7 @@ Configuration variables:
   Set to ``""`` to remove the default entity category.
 - **time_id** (**Required**, :ref:`config-id`): The ID of the time entity. Automatically set
   to the ID of a time component if only a single one is defined.
+- If Webserver enabled, ``web_server_sorting_weight`` can be set. See :ref:`Webserver Entity Sorting <config-webserver-sorting>`.
```
Comment (jesserockz) [inline components/datetime/index.rst]:
```suggestion
- If Webserver enabled, ``web_server_sorting_weight`` can be set. See :ref:`Webserver Entity Sorting <config-webserver-sorting>`.
```

File: components/alarm_control_panel/index.rst [hunk 10]
```diff
@@ -49,6 +49,8 @@ Configuration variables:
 - **on_chime** (*Optional*, :ref:`Action <config-action>`): An automation to perform
   when a zone has been marked as chime in the configuration, and it changes from closed to open. 
   See :ref:`alarm_control_panel_on_chime_trigger`.
+- If Webserver enabled, ``web_server_sorting_weight`` can be set. See :ref:`Webserver Entity Sorting <config-webserver-sorting>`.
+
```
Comment (jesserockz) [inline components/alarm_control_panel/index.rst:53]:
```suggestion
```

**Review by jesserockz — CHANGES_REQUESTED** [review]
Trailing whitespace =)

**Review by jesserockz — APPROVED** [review]
